src/views/AddItem.jsx

```diff
@@ -1,31 +1,76 @@
 import { useState } from 'react';
 import { addItem } from '../api/firebase';
 
-export function AddItem({ listToken }) {
+export function AddItem({ listToken, data }) {
 	const [formData, setFormData] = useState({
 		itemName: '',
 		daysUntilNextPurchase: 7,
 	});
 
 	const [message, setMessage] = useState('');
+	const [duplicateError, setDuplicateError] = useState(false);
 
 	const { itemName, daysUntilNextPurchase } = formData;
 
+	const specialChar = [
+		'$',
+		'&',
+		'+',
+		',',
+		':',
+		';',
+		'=',
+		'?',
+		'@',
+		'#',
+		'|',
+		'<',
+		'>',
+		'.',
+		'^',
+		'*',
+		'(',
+		')',
+		'%',
+		'!',
+		'-',
+	];
+
+	const isDuplicate = data.some(
+		(item) =>
+			item.name
+				.toLowerCase()
+				.split('')
+				.filter((x) => !specialChar.includes(x) && x !== ' ')
+				.join('') ===
+			itemName
+				.toLowerCase()
+				.split('')
+				.filter((x) => !specialChar.includes(x) && x !== ' ')
+				.join(''),
+	);
+
```

Comment:
There must be another way to do this

Reply:
I am so surprised that there is not a special character method already! On a cellphone, there are so many more special characters eg money, math, copyright etc as well as international users would have different keyboards. I agree with you Pat. It does feel like there should be a more encompassing way.